docs/contributing.md

```diff
@@ -0,0 +1,26 @@
+
+We welcome proposals for longer topic-based guides. Suggest them as an [issue in the documentation repo](https://github.com/opensafely/documentation/issues), and we'll work with you to define an overall structure and format, before you write and submit the draft.
+
+## Reusable actions
+These are units of software that solve a problem for several studies without the need to copy-and-paste between them, [described in more detail here](./actions-reusable.md).  They can be shared between researchers, even between groups that use different programming languages, and are one of the best ways you can make contributions that benefit the community. If you've written a reusable action you'd like to contribute to the actions library, please get in touch at [team@opensafely.org](mailto:team@opensafely.org).
+
+## Study definition patterns
+[Study definitions](./study-def.md) are intended to be shared under open source licences, and they can be found in the [OpenSAFELY Github organisation](https://github.com/opensafely).  We welcome all ideas that may be able to help other members of the community, which we can incorporate in our documentation ([for example on this page about study definition tips](https://docs.opensafely.org/study-def-tricks/)); we're also happy to help publicise blog posts, or host them as guest blog posts on our website. Just submit a ticket to the [documentation repository](https://github.com/opensafely/documentation/issues) with your suggestion.
+
+## Peer support
+
+We encourage researchers to post questions in the [Q&A Forum](https://github.com/opensafely/documentation/discussions). We would love more people to chip in and attempt to answer questions!
+
+## Core code
+We would like to accept _ad hoc_ contributions to the [core code](https://github.com/opensafely-core/), but this is relatively hard for us to do. New features and bugfixes need a lot work from our side to integrate and maintain, because a lot of the design and prioritisation decisions are made with lots of different parts of the framework in mind. All but the most trivial changes also come with intellectual property considerations, so we also need to secure Contributor Agreements from individuals or their employers.  We plan to seek funding to help resource this kind of work, as we believe it can really strengthen and build our community, but for the time being, we suggest [filing bug reports or feature requests](https://github.com/opensafely-core) rather than contributing code.
+
+An exception is where external contributors are in a position to dedicate significant resource (e.g. more than one full-time person equivalent) for a prolonged period of time. In these cases, we will be happy to consider allocating resource from our side to help with the planning and integration work; just get in touch with your proposed features at [team@opensafely.org](mailto:team@opensafely.org).
```

Comment:
> consider allocating resource

Very small point, but I try to avoid this word when referring to people. Could resource be changed to "someone" or "a person" instead?

Reply:
Here's my go at a slight rewrite for conciseness:

```suggestion
Where external contributors can dedicate significant time and effort (e.g. more than one full-time person equivalent), we will be happy to consider helping you with the planning and integration work. Get in touch with your proposed features at [team@opensafely.org](mailto:team@opensafely.org).
```